core/app/models/spree/variant/options_presentation.rb

```diff
@@ -0,0 +1,47 @@
+class Spree::Variant::OptionsPresentation
+  WORDS_CONNECTOR = ', '.freeze
+
+  attr_reader :variant
+
+  def initialize(variant)
+    @variant = variant
+  end
+
+  def to_sentence
+    options = option_values
+    options = sort_options(options)
+    options = present_options(options)
+
+    join_options(options)
+  end
+
+  private
+
+  def option_values
+    variant.option_values.includes(:option_type).to_a
+  end
+
+  def sort_options(options)
+    options.sort_by { |o| o.option_type.position }
+  end
+
+  def present_options(options)
+    options.map do |ov|
+      method = "present_#{ov.option_type.name}_option"
```

Comment:
There's only one case when a different method than `present_option` is used (color option value) - maybe we could use a simple `if` condition?

Reply:
We could but my current implementation allows to handle any other option types by defining additional methods only and not changing existing code